solutions/secure/variables.tf

```diff
@@ -0,0 +1,177 @@
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API Token"
+  sensitive   = true
+}
+
+variable "region" {
+  description = "The IBM Cloud region where the instance of IBM Cloud Object Storage(COS) is provisioned."
+  type        = string
+}
+
+variable "create_cos_instance" {
+  description = "Set as true to create a new IBM Cloud Object Storage(COS) instance."
+  type        = bool
+  default     = true
+}
+
+variable "existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group in which Cloud Object Storage(COS) instance will be provisioned."
+}
```

Comment:
If resource group name is not provided, the module is creating a new resource group. Then why it is required to be a mandatory variable?

Reply:
Resource group name will always be needed, whether its created or not is defined by the `existing_resource_group` boolean.